bootstrap/mod/src/main/java/org/geysermc/geyser/platform/mod/GeyserModBootstrap.java

```diff
@@ -85,6 +85,15 @@ public void onGeyserInitialize() {
         }
         this.geyserLogger.setDebug(geyserConfig.isDebugMode());
         GeyserConfiguration.checkGeyserConfiguration(geyserConfig, geyserLogger);
+
+        if (!ModConstants.isModernVersion() && !this.platform.testViaPresent(this)) {
+            this.geyserLogger.error("-------------------------------------------------------------------------");
+            this.geyserLogger.error("The current server version is not supported unless ViaVersion is present!");
+            this.geyserLogger.error("Geyser will not start unless you add ViaVersion!");
+            this.geyserLogger.error("-------------------------------------------------------------------------");
+            return;
+        }
```

Comment:
We could also throw a runtime exception :p
Or, alternatively, make the older adapters depend on via? dunno if that'd work on neo, I guess we'll see